pkg/engine/engine.go

```
@@ -194,6 +196,11 @@ func (e Engine) initFunMap(t *template.Template) {
 	funcMap := funcMap()
 	includedNames := make(map[string]int)
 
+	// Set custom template funcs
+	for k, v := range e.CustomTemplateFuncs {
+		funcMap[k] = v
```

This will not allow overriding existing functions (a potential use case), only adding new functions. Do we want to take the approach #12135 takes?:

https://github.com/helm/helm/pull/12135/files#diff-9a4b0f64963e54937214c78c91c3d404eaef74b571e89f370ed7de3c9ff86ca1R215

---

It allows to override functions created by the "funcMap" function(check test case, it overrides the "upper" function), overriding others embedded funcs like tmpl, include and etc seems wrong

---

(and on the OTOH, if we don't want to override these functions, right now we have a awkward interface where the user will attempt, and Helm will silently fail. User would literally have to come read the source code to understand the problem)

---

ie. if we don't want to allow overriding  tmpl, include, etc. We should add an error if the user attempts to do so (rather than silently failing)

---

Ok, I agree. I will move below to allow override all functions